wrappers/python/tests/TestCharmmFiles.py

```diff
@@ -361,6 +361,40 @@ def test_Nonbonded_Exclusion(self):
         # Compare with value computed with NAMD.
         self.assertAlmostEqual(energy, -2154.5539, delta=1e-3*abs(energy))
 
+    def test_Constraints(self):
+        """Test that bond and angles constraints are correctly added into the system"""
+        psf = CharmmPsfFile('systems/water_methanol.psf')
+        params = CharmmParameterSet('systems/water_methanol.prm')
+        # the system is made of one water molecule and one methanol molecule
+        hBonds_water = [[0, 1], [1, 2]]
+        hAngles_water = [[0, 2]]
+        hBonds_methanol = [[3, 4], [3, 5], [3, 6], [7, 8]]
+        allBonds_methanol = hBonds_methanol + [[3, 7]]
+        hAngles_methanol = [[4, 5], [4, 6], [5, 6], [3, 8]]
+        system = psf.createSystem(params, constraints=None, rigidWater=False)
+        self.assertEqual(system.getNumConstraints(), 0)
+        system = psf.createSystem(params, constraints=None, rigidWater=True)
+        self.assertCountEqual([system.getConstraintParameters(i)[:2] for i in range(system.getNumConstraints())],
```

Comment:
Instead of just checking the number of constraints, how about verifying that the two lists of constraints are the same?  Alternatively you could call `validateConstraints()`: https://github.com/openmm/openmm/blob/master/wrappers/python/tests/validateConstraints.py

Reply:
assertCountEqual means they have the same elements. Indeed the name is very misleading, and it doesn't work under python2. I changed it to assertEqual on the sorted lists

Reply:
That is indeed a very misleading name!  Python 2 isn't important.  We've dropped support for it.